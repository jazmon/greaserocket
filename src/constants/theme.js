// @flow
import color from 'color';

const theme = {
  DARK_PRIMARY_COLOR: '#388E3C',
  DEFAULT_PRIMARY_COLOR: '#4CAF50',
  LIGHT_PRIMARY_COLOR: '#C8E6C9',
  TEXT_PRIMARY_COLOR: '#FFFFFF',
  ACCENT_COLOR: '#FFC107',
  PRIMARY_TEXT_COLOR: 'rgba(0, 0, 0, 0.87)',
  SECONDAY_TEXT_COLOR: 'rgba(0, 0, 0, 0.54)',
  DIVIDER_COLOR: 'rgba(0, 0, 0, 0.12)',
  DISABLED_TEXT_COLOR: 'rgba(0, 0, 0, 0.38)',
  WHITE: '#FFF',
  BLACK: '#000',
};

export default {
  ...theme,
  LIGHT_STATUS_BAR: color(theme.DARK_PRIMARY_COLOR).dark(),
};
