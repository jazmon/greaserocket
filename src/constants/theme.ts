// @flow
import color from 'color';

const theme = {
  colors: {
    primary: {
      normal: '#4CAF50',
      dark: '#388E3C',
      light: '#C8E6C9',
    },
    accent: {
      normal: '#FFC107',
    },
    white: '#FFFFFF',
    black: '#000000',
    transparent: 'transparent',
  },
  text: {
    colors: {
      light: '#FFFFFF',
      primary: 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 0, 0, 0.54)',
      disabled: 'rgba(0, 0, 0, 0.38)',
      divider: 'rgba(0, 0, 0, 0.12)',
    },
    size: 12,
  },
};

export type ThemeType = typeof theme;

export function getStatusBarTextColor(
  backgroundColor: string = 'light-content',
): 'light-content' | 'dark-content' {
  return color(backgroundColor).dark() ? 'light-content' : 'dark-content';
}

export default theme;
