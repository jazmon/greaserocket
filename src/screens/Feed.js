// @flow
import React from 'react';
import {
  View,
  Text,
  ListView,
  StyleSheet,
} from 'react-native';

import Story from '../components/Story';

const data = [
  {
    id: 'story-0',
    text: 'Hello, World!',
    author: {
      id: 'author-0',
      profilePictureUrl: 'http://loremflickr.com/320/240/dog?asdf=0',
      name: 'Jack Johnson',
    },
  },
  {
    id: 'story-1',
    text: 'I like trains',
    author: {
      id: 'author-1',
      profilePictureUrl: 'http://loremflickr.com/320/240/dog?foo=1',
      name: 'John Jackson',
    },
  },
];

const rowHasChanged = (r1, r2) => r1.id !== r2.id;

const ds = new ListView.DataSource({ rowHasChanged });

type Props = {
  navigation: Object,
  router: Object,
};

type State = {
  text: string,
};

class Feed extends React.Component {

  static navigationOptions = {
    title: 'Feed',
  };

  props: Props;
  dataSource: Object;

  constructor(props: Props) {
    super(props);

    this.state = {
      text: 'Feed',
    };

    this.dataSource = ds.cloneWithRows(data);
  }

  state: State;

  renderRow = (story: Object) => (
    <Story {...story} key={story.id} />
  );

  renderSeparator = (sID: string, rID: string) => (
    <View style={styles.separator} key={`${sID}-${rID}`} />
  );

  render() {
    const { text } = this.state;
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{text}</Text>
        <ListView
          dataSource={this.dataSource}
          renderRow={this.renderRow}
          renderSeparator={this.renderSeparator}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#fff',
  },
  text: {
    color: '#000',
    fontSize: 16,
  },
  separator: {
    height: 8,
  },
});

export default Feed;
