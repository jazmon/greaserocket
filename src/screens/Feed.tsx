// @flow
import React from 'react';
import {
  View,
  ListView,
  ScrollView,
  RefreshControl,
  Platform,
  ActivityIndicator,
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { Story as StoryType } from 'types';
import { ReduxState } from 'redux/modules';
import styled from 'styled-components/native';

import { fetchStories, refetchStories } from 'redux/modules/feed';

import Story from 'components/Story';
import PlaceholderStory from 'components/PlaceholderStory';
import Base from 'components/Base';
import { ThemeType } from 'constants/theme';

interface HasID {
  id: number | string;
}
const rowHasChanged = (r1: HasID, r2: HasID): boolean => r1.id !== r2.id;

const ds = new ListView.DataSource({ rowHasChanged });

type Props = {
  // navigation: Object,
  // router: Object,
  fetchStories: () => void;
  refetchStories: () => void;
  error: boolean;
  loading: boolean;
  refetching: boolean;
  stories: Array<StoryType>;
};

const LoadingContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: ${({ theme }: { theme: ThemeType }) =>
    theme.colors.transparent};
`;
const Separator = styled.View`height: 8;`;
const Container = styled.View`
  flex: 1;
  margin-top: ${Platform.OS === 'ios' ? 24 : 0};
  flex-direction: column;
  background-color: ${({ theme }: { theme: ThemeType }) => theme.colors.white};
`;

class Feed extends React.Component<Props, void> {
  static navigationOptions = {
    tabBarLabel: 'Feed',
    tabBarIcon: ({ tintColor }) =>
      <Icon name="ios-paper" size={30} color={tintColor} />,
  };

  static defaultProps = {
    stories: [],
  };

  dataSource: Object;

  constructor(props: Props) {
    super(props);

    this.dataSource = ds.cloneWithRows(props.stories);
  }

  componentDidMount() {
    // this.props.fetchStories();
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.stories) {
      this.dataSource = ds.cloneWithRows(nextProps.stories);
    }
    // if (nextProps.token && !this.props.token) {
    //   this.props.fetchStories();
    // }
    if (nextProps.error) {
      console.log(
        nextProps.error.message || 'error fetching data for feed',
        nextProps.error.stack,
      );
      // alert(nextProps.error.message || 'Error!');
    }
  }

  onRefresh = () => {
    this.props.refetchStories();
  };

  renderRow = (story: StoryType) => <Story story={story} key={story.id} />;

  renderSeparator = (sID: string, rID: string) =>
    <Separator key={`${sID}-${rID}`} />;

  renderLoading = () =>
    <LoadingContainer>
      <ActivityIndicator animating size="large" />
    </LoadingContainer>;

  render() {
    const { loading, refetching } = this.props;
    return (
      <Base>
        <Container>
          {!loading &&
            <ListView
              dataSource={this.dataSource}
              pageSize={6}
              renderRow={this.renderRow}
              renderSeparator={this.renderSeparator}
              enableEmptySections
              renderSectionHeader={(sectionData, sectionId) =>
                <View key={`sh-${sectionId}`} />}
              refreshControl={
                <RefreshControl
                  refreshing={refetching}
                  onRefresh={this.onRefresh}
                />
              }
            />}
          {loading &&
            <ScrollView>
              {[1, 2, 3, 4, 5, 6].map(a =>
                <PlaceholderStory key={String(a)} />,
              )}
            </ScrollView>}
        </Container>
      </Base>
    );
  }
}

const mapState = ({ feed }: ReduxState) => ({
  ...feed,
  // token: user.token,
});

const mapActions = (dispatch: Dispatch<ReduxState>) => ({
  fetchStories: () => dispatch(fetchStories()),
  refetchStories: () => dispatch(refetchStories()),
});

export const FeedComponent = Feed;
export default connect(mapState, mapActions)(Feed);
