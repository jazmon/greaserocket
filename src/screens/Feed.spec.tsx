import 'react-native';
import * as React from 'react';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'constants/theme';
import { FeedComponent as Feed } from './Feed';

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Feed fetchStories={jest.fn} stories={[]} loading={false} />
    </ThemeProvider>,
  );
  expect(tree).toMatchSnapshot();
});
