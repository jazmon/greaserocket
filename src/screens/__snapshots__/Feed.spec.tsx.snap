// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "marginTop": 24,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      enableEmptySections={true}
      pageSize={6}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={undefined}
        />
      }
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      renderSectionHeader={[Function]}
      renderSeparator={[Function]}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
</View>
`;
